TaylorSource/CoreData/NSFRCUpdateHandler.swift

```diff
@@ -0,0 +1,115 @@
+import UIKit
+import CoreData
+
+private class WeakObserver {
+    private(set) weak var value: NSFRCIndexedUpdateConsumer?
+
+    init(_ observer: NSFRCIndexedUpdateConsumer) {
+        value = observer
+    }
+}
+
+public class NSFRCUpdateHandler: NSObject, NSFetchedResultsControllerDelegate {
+    
+    deinit {
+        self.observers.removeAll()
+    }
+    
+    public func addUpdateObserver(observer: NSFRCIndexedUpdateConsumer) {
+        self.observers.append(WeakObserver(observer))
+    }
+    
+    func fullUpdate() {
+        self.sendUpdate(.FullUpdate)
+    }
+    
+    // MARK: Private
+    
+    private var observers = [WeakObserver]()
+    
+    private func sendUpdate(update: NSFRCIndexedUpdate) {
+        self.observers = self.observers.filter { $0.value != nil } // Remove orphaned observers
+        self.observers.forEach { $0.value?.handleIndexedUpdate(update) }
+    }
+    
+    private func createUpdateFromCollectedValues() -> NSFRCIndexedUpdate {
+        let insertedSections = self.insertedSections.copy() as! NSIndexSet
+        let deletedSections = self.deletedSections.copy() as! NSIndexSet
+        let update: NSFRCIndexedUpdate = .DeltaUpdate(
+            insertedSections: insertedSections,
+            deletedSections: deletedSections,
+            insertedRows: self.insertedRows,
+            updatedRows: self.updatedRows,
+            deletedRows: self.deletedRows
+        )
+        return update
+    }
+    
+    private func clearCollectedValues() {
+        self.insertedSections = nil
+        self.deletedSections = nil
+        self.insertedRows = nil
+        self.updatedRows = nil
+        self.deletedRows = nil
+    }
+    
+    // MARK: NSFetchedResultsControllerDelegate
+    
+    private var insertedSections: NSMutableIndexSet!
+    private var deletedSections: NSMutableIndexSet!
+    private var insertedRows: [NSIndexPath]!
+    private var updatedRows: [NSIndexPath]!
+    private var deletedRows: [NSIndexPath]!
+    
+    public func controllerWillChangeContent(controller: NSFetchedResultsController) {
+        self.insertedSections = NSMutableIndexSet()
+        self.deletedSections = NSMutableIndexSet()
+        self.insertedRows = []
+        self.updatedRows = []
+        self.deletedRows = []
+    }
```

Comment:
I agree that it's less than pretty. I've added a private struct to package up the pending changes. I opted for adding it as a `var` since I felt that keeping everything strictly immutable was somewhat cumbersome. If you feel strongly about keeping things immutable I'll change it.